Linux: Bump to 4.3

lrusak opened the pull request:
Another large PR.

Please review and check for errors!

 - [x] Generic
 - [x] Nvidia_Legacy
 - [x] RPi
 - [x] RPi2

MilhouseVH commented:
I've built Nvidia_Legacy with the following [linux.x86_64.conf](http://nmacleod.com/public/oebuild/patches/00_config_43rc5_legacy.txt). It does require #4385. All works fine.

lrusak commented:
@MilhouseVH thanks for pointing that out, I've picked the nvidia_legacy commits from master as well.

MilhouseVH commented:
You'll need to add the Nvidia_Legacy kernel config patch too, as this PR only updates Generic.

fritsch commented:
rc6 is already out. @stefansaraev just puts all the bits together - including SKL firmware that we need for skylake anyways. 

@lrusak please pick from seo again and force push then please.

stefansaraev commented on projects/Generic/linux/linux.x86_64.conf:
```diff
@@ -676,6 +688,7 @@ CONFIG_X86_DEV_DMA_OPS=y
 CONFIG_PMC_ATOM=y
 CONFIG_NET=y
 CONFIG_COMPAT_NETLINK_MESSAGES=y
+CONFIG_NET_INGRESS=y
```
meh. disable this ;)

stefansaraev commented on projects/Generic/linux/linux.x86_64.conf:
```diff
@@ -749,6 +762,7 @@ CONFIG_BRIDGE_NETFILTER=m
 #
 # Core Netfilter Configuration
 #
+CONFIG_NETFILTER_INGRESS=y
```
and this too.

lrusak commented:
rebase and bumped to rc6

stefansaraev commented:
cc people who maintain patches:
@sraue 
hid_sony-add_autorepeat_for_PS3_remotes

@MattDevo 
xpad-fix_xbox360_wireless

@dalehamel
STK1160-addFramescaling

@vpeter4
CX24120-13Z_frontend
saa716x_PCIe_interface_chipset (not yours but you volunteered last time it was broken)

ngene-octopus - I did not add this initialy, but maintained it for well, 2 years ? I have little motivation atm  so if http://linuxtv.org/hg/~endriss/media_build_experimental is not updated by the time this PR is ready for merging - feel free to drop the patch. or ask someone to take over.

fritsch commented:
@lrusak: All is fine - the project counts - not the person that starts initial patches :-)

As my internet connection here is still limited I am very thankful for you guys picking it up.

fritsch commented:
And please pick the hack of @MilhouseVH then after correcting @stefansaraev notes - we should be good to go.

Btw. you also picked the intel "Limited Range" kernel patch. It does not hurt at all - as it does nothing by default, just that you know.

MilhouseVH commented:
RPi/RPi2 config options here: https://github.com/MilhouseVH/OpenELEC.tv/commit/d320cd041940c03d88fb806215423ba79fce4335

Tested and working on a RPi Model B and RPi2, with [rpi-4.3.y](https://github.com/raspberrypi/linux/compare/raspberrypi:7379047d5585187d1288486d4627873170d0005a...raspberrypi:eb73ce4c8c58269923fae702b80641f9b4f88751).

vpeter4 commented:
CX24120-13Z_frontend - can be removed (already in kernel)
saa716x_PCIe_interface_chipset - leave as todo file (will look one day)

popcornmix commented:
@MilhouseVH are you sure 4.3 is stable on pi2. 4.3-rc5 wasn't stable for me (with raspbian). Generally got kernel panics after a couple of minutes (sometimes during boot). pi1 seemed okay.
It's possible rc6 has fixed something. It's possible OE is not using a .config option that is problematic. But it's also possible you will see instability if you run for longer.

MilhouseVH commented:
@popcornmix: With 4.3-rc6 and the aforementioned configs, both RPi1 and RPi2 have been up and stable for 11-12 hours (still playing videos, nothing nasty in dmesg or journalctl).

The first time I tried to build 4.3-rc6 for RPi2 I used the existing 4.1.8 config, ie. no changes, as I expected the kernel to choke and halt the build wanting answers for new options but it succeeded, so presumably it used all defaults. This build panicked on boot, so I ran `make oldconfig` on the new 4.3-rc6 kernel and accepted most defaults but changed the two I mentioned - maybe `CPU_SW_DOMAIN_PAN` is the key?

Now with the updated config, 4.3-rc6 is working well. Can you reproduce or maybe I just got lucky?

lrusak commented:
I have updated the rpi support patches and added the kernel configs from @MilhouseVH 

changed the two options mentioned by @stefansaraev 

@fritsch for the intel patch how is it enabled? via xrandr or xorg.conf or something else?

Anything else outstanding other than some .todo patches?

popcornmix commented:
Let me try with rc6 and raspbian and if that fails I'll take a closer look in the difference in our configs. Sounds promising that it's been stable for 11 hours, so it's either an upstream fix in rc6 or you've avoided the bad config option I have.

popcornmix commented:
Okay, I should have paid attention to @MilhouseVH. The kernel option that causes the crashing on Pi is CPU_SW_DOMAIN_PAN. With that disabled all seems fine.

dalehamel commented:
The patch I submitted was mainlined, so no point in having it here.

On Wed, Oct 21, 2015 at 6:38 PM, popcornmix <notifications@github.com>
wrote:

> Okay, I should have paid attention to @MilhouseVH
> <https://github.com/MilhouseVH>. The kernel option that causes the
> crashing on Pi is CPU_SW_DOMAIN_PAN. With that disabled all seems fine.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenELEC/OpenELEC.tv/pull/4386#issuecomment-150044302>
> .
>

fritsch commented:
@lrusak: Yes - via xrandr. It does not harm anything as it is not used besides a user would be setting it explicitely. Furthermore it does not make sense with mainline kodi as the default method with preferred VAAPI Render will scale to full range anyways. That being said: decide what you want to do :-)

lrusak commented:
Rebased and cleaned up unwanted patches.

Any other objections?

MilhouseVH commented:
All 4 build here, RPi, RPi2 and Nvidia_Legacy tested without problems (unable to test Generic due to lack of hardware).

fritsch commented:
RC7 is already out. two more patches can be dropped: https://github.com/fritsch/OpenELEC.tv/commit/0ce2c8c8e76e2274350ef96fab08591a3e793581

lrusak commented:
Bumped to 4.3-rc7 and removed unneeded patches.

lrusak commented:
Rebased and updated to 4.3 final. Had to fix the kernel configs but should be good to go now.

fritsch commented:
I would squash https://github.com/lrusak/OpenELEC.tv/commit/362f1f09d260d2f2d8b0f0a573a5e13463ae5096 and https://github.com/lrusak/OpenELEC.tv/commit/e8f8adda76bf00b6a65ed0a1768ce41031080686

into https://github.com/lrusak/OpenELEC.tv/commit/3290c64e08b588d9758c09fe7bfd1d39127fdac4

That way we can more easily revert if needed.

lrusak commented:
Thanks for pointing that out @fritsch

fritsch commented:
Something else: master is already on 4.1.12 - so one needs to think if oe-next is first rebased to master and then this PR here is again rebased.

sraue commented:
added in https://github.com/OpenELEC/OpenELEC.tv/commit/10735c6bfb443810e32f3b925ace768eb9dbe254 https://github.com/OpenELEC/OpenELEC.tv/commit/40765be557cd9ba3bdc6753c4a2fba2e8932d993 https://github.com/OpenELEC/OpenELEC.tv/commit/3208b638f50c3602219cb4a0fd5785c89afdceb4 https://github.com/OpenELEC/OpenELEC.tv/commit/2ebf4707ce1fde20b44a182b8b87280d30f6797a https://github.com/OpenELEC/OpenELEC.tv/commit/497312cdff2954e08a0f6ca1a5974ead01162f3d https://github.com/OpenELEC/OpenELEC.tv/commit/b8f09d566381df814c524fe33d831b1f1eb95670 https://github.com/OpenELEC/OpenELEC.tv/commit/57b7916473da751bdbbe4f5bf72cbe84ef2c951b https://github.com/OpenELEC/OpenELEC.tv/commit/ffac260ff295cce4af7d8e185942e61bb83f8ee2
